### PR TITLE
commit prueba Maria

### PR DESCRIPTION
**Pull request by maryangrb**
Haciendo el Pull Request.

### CODE REVIEW

**Comment by maryangrb** [comment]
Subiendo de nuevo

File: src/test/java/dataProvider.java [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+        cookies.click();
+        Thread.sleep(2000);
+    }
+
+    @AfterMethod
+    public void tearDown(){
+        driver.close();
+    }
+    @DataProvider(name = "emails")
+    public Object[][] crearEmails(){
```
Comment (seleniumcurso) [inline src/test/java/dataProvider.java:38]: esto se podria haber creado en otra clase

File: src/test/java/dataProvider.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    @Test(dataProvider = "emails")
+    public void dataProviderEmailTest (String email) throws InterruptedException {
+
+        WebElement usernameLoginField = driver.findElement(By.id("username"));
+        usernameLoginField.sendKeys(email);
+
+        WebElement passwordLoginField = driver.findElement(By.id("password"));
+        passwordLoginField.sendKeys("holamundo”.");
+
+        WebElement buttonLoginIn = driver.findElement(By.xpath("//*[@id=\"login-form\"]/fieldset/div[5]/button"));
```
Comment (seleniumcurso) [inline src/test/java/dataProvider.java:56]: no utilizar xpath absolutos

File: src/test/java/dataProvider.java [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+        buttonLoginIn.click();
+
+        Thread.sleep(2000);
+
+        WebElement smsError = driver.findElement(By.xpath("//*[@id=\"login-form\"]/fieldset/div[1]/div/div/div[2]/p"));
```
Comment (seleniumcurso) [inline src/test/java/dataProvider.java:61]: no utilizar xpath absolutos

File: src/test/java/prueba_mailchimp.java [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+        Assert.assertTrue((texto.getText().contains("Need a Mailchimp account?")), "El sitio contiene el texto");
+    }
+
+    //Caso de prueba 3:
+    @Test (priority = 3)
+    public void loginErrorTest(){
+        WebElement usernameLoginField = driver.findElement(By.id("username"));
+        usernameLoginField.sendKeys("XXXXX@gmail.com");
+
+        WebElement buttonLoginIn = driver.findElement(By.xpath("//*[@id=\"login-form\"]/fieldset/div[4]/button"));
```
Comment (seleniumcurso) [inline src/test/java/prueba_mailchimp.java:63]: no utilizar xpath absolutos

File: src/test/java/prueba_mailchimp.java [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+        buttonLoginIn.click();
+
+        WebElement smsError = driver.findElement(By.xpath("//*[@id=\"empty-error\"]/div/div/div[2]/p"));
```
Comment (seleniumcurso) [inline src/test/java/prueba_mailchimp.java:66]: no utilizar xpath absolutos

**Comment by seleniumcurso** [comment]
Te he marcado algunas mejoras en el codigo. Nunca usar xpath absolutos.
Sin ser por eso, el examen esta muy bien! Se demuestra dominio de Selenium
Felicitaciones! El examen está aprobado

File: src/test/java/dataProvider.java [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+import org.testng.Assert;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+import utilities.GetProperties;
+
+import java.util.concurrent.TimeUnit;
+
+public class dataProvider {
```
Comment (ellieMayVelasquez) [inline src/test/java/dataProvider.java:14]: I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [Nm: Class names should start with an upper case letter (NM_CLASS_NAMING_CONVENTION)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#nm-class-names-should-start-with-an-upper-case-letter-nm-class-naming-convention).
Class names should be nouns, in mixed case with the first letter of each internal word capitalized. Try to keep your class names simple and descriptive. Use whole words-avoid acronyms and abbreviations (unless the abbreviation is much more widely used than the long form, such as URL or HTML).

**Pull request merged**